nef/Extensions/AppDelegate+Notifications.swift

```diff
@@ -0,0 +1,112 @@
+
+import AppKit
+import UserNotifications
+import Bow
+import BowEffects
+
+extension AppDelegate: UNUserNotificationCenterDelegate {
+    func isLocalNotification(_ aNotification: Notification) -> Bool {
+        guard let userInfo = aNotification.userInfo,
+              let launchOption = userInfo["NSApplicationLaunchIsDefaultLaunchKey"] as? Int else { return false }
+        
+        return launchOption == 0
+    }
+    
+    func registerNotifications() {
+        NefNotification.center.delegate = self
+        NefNotification.center.requestAuthorization(options: [.alert, .sound]) { granted, _  in }
+    }
+    
+    func removeOldNotifications() {
+        NefNotification.center.removeAllDeliveredNotifications()
+    }
+    
+    func showNotification(title: String, body: String, imageData: Data? = nil, actions: [NefNotification.Action] = [], id: String = UUID().uuidString) {
+        let content = UNMutableNotificationContent()
+        content.title = title
+        content.body = body
+        content.categoryIdentifier = id
+        
+        if let data = imageData {
+            content.userInfo = [NefNotification.UserInfoKey.imageData: data]
+        }
+        
+        let category = UNNotificationCategory(identifier: id,
+                                              actions: actions.map(\.unNotificationAction),
+                                              intentIdentifiers: [],
+                                              hiddenPreviewsBodyPlaceholder: "",
+                                              options: .customDismissAction)
+        
+        let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 0.5, repeats: false)
+        let request = UNNotificationRequest(identifier: id, content: content, trigger: trigger)
+        
+        NefNotification.center.setNotificationCategories([category])
+        NefNotification.center.add(request)
+    }
+    
+    // MARK: delegate <UNUserNotificationCenterDelegate>
+    func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+        completionHandler(.alert)
+    }
+    
+    func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
+        guard let userInfo = response.notification.request.content.userInfo as? [String: Any] else { return }
+        command = .notification(userInfo: userInfo, action: response.actionIdentifier)
+        applicationDidFinishLaunching(Notification(name: .NSThreadWillExit))
+        completionHandler()
+    }
+    
+    func processNotification(_ userInfo: [String: Any], action: String) -> IO<AppDelegate.Error, Either<Void, URL>> {
+        guard let image = userInfo[NefNotification.UserInfoKey.imageData] as? Data else { return IO.raiseError(.notification)^ }
```

Comment:
Since the error is an `AppDelegate.Error`, don't you think `.missingData` is too generic? it could be any kind of data, happening anywhere in the code. Should I create a `NefNotification.Error` enum?